tests/module_1/test_meteo_api.py

```diff
@@ -1,7 +1,62 @@
-""" This is a dummy example to show how to import code from src/ for testing"""
+from src.module_1.module_1_meteo_api import get_data_from_api, process_data
+import pytest  # noqa: F401
+from typing import Any
 
-from src.module_1.module_1_meteo_api import main
+API_URL = "https://archive-api.open-meteo.com/v1/archive"
+COORDINATES = {"Madrid": {"latitude": 40.416775, "longitude": -3.703790}}
+VARIABLES = ["temperature_2m_mean"]
+START_DATE = "2010-01-01"
+END_DATE = "2010-01-02"
 
 
-def test_main():
-    raise NotImplementedError
+def test_get_data_from_api_success() -> None:
+    params = {
+        "start_date": START_DATE,
+        "end_date": END_DATE,
+        "latitude": COORDINATES["Madrid"]["latitude"],
+        "longitude": COORDINATES["Madrid"]["longitude"],
+        "daily": ",".join(VARIABLES),
+    }
+    r = get_data_from_api(API_URL, params)  # noqa: F841
```

Comment:
You don't need to call the API to check that everything is working. Take a look at the concept of Mocking in tests. We fake that the API is there and gives a pre-defined response. Next, we can check the rest of the code with that fake data.